test/e2e/itn_e2e_test.go

```diff
@@ -0,0 +1,288 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package itn_e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// coupled with Makefile
+const APP_PATH = "../../build/spot-itn"
+const TEST_REGION = "us-west-2"
+const API_RETRY_COUNT = 5
+const RETRY_SLEEP_SEC = 7
+
+var ctx = context.Background()
+
+func TestSpotITN(t *testing.T) {
+	// Pre-checks
+	_, err := os.Stat(APP_PATH)
+	require.Nil(t, err)
+
+	// Setup
+	cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(TEST_REGION))
+	require.Nil(t, err)
+	ec2Client := ec2.NewFromConfig(cfg)
+	require.NotNil(t, ec2Client)
+
+	launchTemplate, ltCleanup := CreateLTForTest(*ec2Client)
+	require.NotNil(t, launchTemplate.LaunchTemplateId)
+	defer ltCleanup()
+	launchTemplateID := *launchTemplate.LaunchTemplateId
+
+	spotInstance, fleetCleanup := CreateSpotInstance(*ec2Client, launchTemplateID)
+	require.NotNil(t, spotInstance.InstanceId)
+	defer fleetCleanup()
+	spotInstanceID := *spotInstance.InstanceId
+
+	// Run spot-interrupter with created Spot instance
+	spotItnCommand := exec.Command(APP_PATH,
+		"--instance-ids", spotInstanceID, "--region", TEST_REGION)
+	spotiOutput, err := spotItnCommand.Output()
+	require.Nil(t, err)
+	spotiOutputClean := string(spotiOutput)
+
+	// Validate expected events happened to the designated instance
+	assert.Contains(t, spotiOutputClean, spotInstanceID)
+	assert.Contains(t, spotiOutputClean, "✅ Rebalance Recommendation sent")
+	// TODO issue: https://github.com/aws/amazon-ec2-spot-interrupter/issues/8
+	// assert.Contains(t, spotiCleanOutput, "⏳ Interruption will be sent in 15 seconds")
+	assert.Contains(t, spotiOutputClean, "✅ Spot 2-minute Interruption Notification sent")
+	assert.Contains(t, spotiOutputClean, "✅ Spot Instance Shutdown sent")
+
+	// Validate Spot instance terminating
+	time.Sleep(time.Second * 5)
```

Comment:
[nit] Curious if this is required along with the RETRY_SLEEP_SEC?

Reply:
Nope, will remove in revision. I had this before adding `|| spotInstance.State.Name == ec2types.InstanceStateNameShuttingDown` check to give it a bit more time.